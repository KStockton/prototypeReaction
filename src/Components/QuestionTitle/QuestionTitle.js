import React, { Component } from 'react' 
export default class QuestionTitle extends Component{
  constructor(props){
    super(props)

  this.state = {
    index: 0,
    showResult: ''
    }
  }


checkSelection = (event) => {
  let showResult;

  if(event.target.value === this.props.topicChoice[this.state.index].correctAnswer) {
    localStorage.removeItem(this.props.topicChoice[this.state.index].id)
    showResult = 'Correct'
  } else {
    showResult = 'Incorrect'
    localStorage.setItem(this.props.topicChoice[this.state.index].id, JSON.stringify(this.props.topicChoice[this.state.index]))
    this.filterLocalStorage()
  }
  this.setState({ index: this.state.index +1, showResult: showResult }, () => setTimeout(() =>{
    this.setState( { showResult: ''})}, 2000 ))
  }
  
  filterLocalStorage () {
    let removeDup = [];

    if(removeDup.length > 0){
      for(let i = 0; i < localStorage.length; i++) {
        removeDup.push(JSON.parse(localStorage.getItem(localStorage.key(i))))
      }
      removeDup.forEach((uniqueId, index) =>{
        if(uniqueId.id === this.props.topicChoice[this.state.index].id){
          removeDup.splice(index, 1)
        }
      })
      localStorage.clear()
      for(let i = 0; i < removeDup.length; i++){
        localStorage.setItem(removeDup[i].id, JSON.stringify(i))
      }
    }
}


componentDidUpdate = () =>{
  if(this.state.index + 1 === this.props.topicChoice.length){
    this.setState({ index: 0 })
  }
}
 
  render() {
      const { index, showResult } = this.state
      let { questions, answers, resource} = this.props.topicChoice[index]

  return (
    <article >
      <h2>{questions}</h2>
        <section className="answer-wrapper">
          { answers.map((answer, index) =>{
           return <button key={index} value={answer} onClick={this.checkSelection}>
                   {answer}
                  </button>
            })
          }
        </section>
        <p className="show-result">{showResult}</p>
      <a href={resource}>Resource</a>
    </article>
 )
}
}
 
