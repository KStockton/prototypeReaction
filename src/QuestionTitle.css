h3 {
 font-size: 35px;
 background-color: #686868;
 color: white;
}