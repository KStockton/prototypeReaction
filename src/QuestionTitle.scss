article h3 {
 font-size: 35px;
 background-color: #686868;
 color: white;
 
}

article h2 {
  margin: 52px 0px;
}
 .answer-wrapper {
   display: flex;
   flex-direction: row;
   justify-content: space-around;
   a {
    display: inline-block;
    border-radius: 8px;
    width: 198px;
    box-shadow: 7px 8px 7px 1px #928686d4, inset -2px -3px 7px 1px #fbfbfb4f;
    background-color: rgb(35, 0, 22);
    color: white;
  }
}

answerText {
 font-size: 1.3rem;
}

 .resource {
   font-size: 30px;
 }