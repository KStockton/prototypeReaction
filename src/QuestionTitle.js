import React, { Component } from 'react' 
import './QuestionTitle.scss'

export default class QuestionTitle extends Component{
  constructor(props){
    super(props)

  this.state = {
    index: 0
    }
  }


checkAnswer = (event) => {
  if(event.target.value === this.props.topicChoice[this.state.index].correctAnswer) {
    console.log('cheers')
  } else {
    localStorage.setItem()
  }
}


  render() {

      const { index } = this.state
      let { questions, answers, resource} = this.props.topicChoice[index]
      
  return(
    <article >
    <h4>{questions}</h4>
    <h5>Select Your Answer</h5>
    <section className="answer-wrapper">
      { answers.map((answer, index) =>{
        return <button key={index} value={answer} onClick={this.checkAnswer}>{answer}</button>
      })
      }
    </section>
      <a href={resource}>Resource</a>
  </article>
 )
}
}
 
