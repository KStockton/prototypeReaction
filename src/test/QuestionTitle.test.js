import React from 'react';
import QuestionTitle from '../QuestionTitle';
import { shallow } from "enzyme";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })



const mockTopicChoice = [
  {
   id: "2932-tq-iwpjt-0oodks",
   category: "Array Prototypes",
   questions: "What does .unshift do to an array ?",
   answers: [
   "adds items to the beginning",
   "removes items from beginning",
   "adds items to the end"
   ],
   correctAnswer: "adds items to the beginning",
   correct: false,
   resource: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift"
  },
  {
   id: "2932-f5-o4cm9-lcfdks",
   category: "Array Prototypes",
   questions: "What array prototype checks if an array has a certain value and returns true or false ?",
   answers: [
   ".findIndex",
   ".lastIndexOf",
   ".includes"
   ],
   correctAnswer: ".includes",
   correct: false,
   resource: "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes"
  }
]



describe('QuestionTitle', () =>{
  let wrapper;

  beforeEach(() =>{
   wrapper = shallow(<QuestionTitle topicChoice={mockTopicChoice}/>
   )
  })
  it('should match the snapshot with all data passed in', () =>{
   expect(wrapper).toMatchSnapshot()
  })
  it('QuestionTitle state should have default properties', () =>{
   expect(wrapper.state().index).toEqual(0)
   expect(wrapper.state().showResult).toEqual('')
  })
  it('When the selection is correct it should display "correct" ', () => {
   wrapper.find('.answer-wrapper').childAt(0).simulate('click', { target: { value: "adds items to the beginning"}})
   expect(wrapper.state().showResult).toEqual("Correct")
   setTimeout(() => {expect(wrapper.state().showResult).toEqual('')}, 1000)
  })
  it('When the selection is incorrect it should display "incorrect', () =>{
   wrapper.find('.answer-wrapper').childAt(1).simulate('click', { target: { value: "adds items to the end"}})
   expect(wrapper.state().showResult).toEqual("Incorrect")
   setTimeout(() => {expect(wrapper.state().showResult).toEqual('')}, 1000)
  })


  

})

