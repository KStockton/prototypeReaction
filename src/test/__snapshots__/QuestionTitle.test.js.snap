// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionTitle should match the snapshot with all data passed in 1`] = `
<article>
  <h2>
    What does .unshift do to an array ?
  </h2>
  <section
    className="answer-wrapper"
  >
    <button
      key="0"
      onClick={[Function]}
      value="adds items to the beginning"
    >
      adds items to the beginning
    </button>
    <button
      key="1"
      onClick={[Function]}
      value="removes items from beginning"
    >
      removes items from beginning
    </button>
    <button
      key="2"
      onClick={[Function]}
      value="adds items to the end"
    >
      adds items to the end
    </button>
  </section>
  <p
    className="show-result"
  />
  <a
    href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift"
  >
    Resource
  </a>
</article>
`;
